.github/workflows/sync.yml

```diff
@@ -0,0 +1,194 @@
+            // for the comparison below. Unfortunately, GitHub's API response does not seem
+            // to contain any reference to the upstream repo of the fork.
+            const botForks = botRepos.flatMap(({ full_name, fork }) => fork ? [full_name.replace(process.env.BOT_USERNAME, 'exercism')] : [])
+
+            const tracks = ${{ steps.fetch-tracks.outputs.result }}
+
+            return tracks.filter(track => !botForks.includes(track))
+
+      - name: Fork repos that haven't been forked yet
+        if: steps.required-forks.outputs.result != '[]'
+        uses: actions/github-script@47f7cf65b5ced0830a325f705cad64f2f58dddf7
+        with:
+          github-token: ${{ secrets.BOT_PERSONAL_ACCESS_TOKEN }}
+          script: |
+            const trackReposFullNames = ${{ steps.required-forks.outputs.result }}
+            const trackRepos = trackReposFullNames.map((fullName) => fullName.split('/')[1])
+
+            for (repo of trackRepos) {
+              await github.repos.createFork({
+                owner: 'exercism',
+                repo
+              })
+
+              // Sleep a bit to avoid rate-limiting
+              await new Promise(x => setTimeout(x, 2000))
+            }
+
+            // Forking repos happens asynchronously on GitHub, therefore
+            // we need to sleep a bit to ensure it has been created
+            await new Promise(x => setTimeout(x, 60000))
+
+  abort-timer:
+    needs: fetch-tracks
+    runs-on: ubuntu-20.04
+    name: ⚠ 5 MINUTES TO ABORT
+    environment: abort-timer
+    steps:
+      - run: exit 0
+
+  push-to-tracks:
+    needs: [fetch-tracks, abort-timer]
+    name: Push to ${{ matrix.track }}
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+
+    # Launch one job per track.
+    # This is probably less efficient than running everything in one job
+    # and manually cloning and checking out the repos. However, launching a job
+    # lets us use actions like actions/checkout.
+    # It also gives us a pretty job overview that makes it easy to spot issues with
+    # particular tracks.
+    strategy:
+      fail-fast: false
+      matrix:
+        track: ${{ fromJson(needs.fetch-tracks.outputs.tracks) }}
+
+    steps:
+      - name: Checkout main repo
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          path: main
+
+      - name: Checkout track repo
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          repository: ${{ matrix.track }}
+          token: ${{ secrets.BOT_PERSONAL_ACCESS_TOKEN }}
+          path: track-repo
+
+      - name: Checkout new branch on track repo
+        id: branch
+        run: |
+          cd track-repo
+
+          # github.run_number is a unique ID for each workflow run in a repo
+          # WARNING: the run_number will NOT be updated if a workflow is re-run
+          branch="🤖/org-wide-files/${{ github.run_number }}"
```

Comment:
Does the emoji branch name work in Windows?

Reply:
Yes, at least in Git Bash. I haven't tried `cmd` or `powershell`

Reply:
Would be worthwhile to check I think

Reply:
Excellent!